docs/content/guides/developer/advanced/congestion-control.mdx

```diff
@@ -0,0 +1,59 @@
+---
+title: Congestion Control
+description: Congestion control is the system that limits the rate of transactions writing to a single shared object, preventing the network from becoming overloaded with checkpoints that take too long to execute.
+---
+
+The Sui network’s object-based architecture allows us to process many different user transactions massively in parallel, in a way that’s not possible on most other networks. However, if multiple transactions are all writing to the same shared object, they must execute in sequential order. There is a limit to how many transactions we can process that touch one specific object.
+
+**Congestion control** is the system that limits the rate of transactions writing to a single shared object, preventing the network from becoming overloaded with checkpoints that take too long to execute.
+
+If you see transactions fail with the error `ExecutionCancelledDueToSharedObjectCongestion`, you are observing congestion control at work! Read this document to learn:
+- How transaction space is allocated on a congested shared object.
+- How to bid for priority access to limited space.
+- How to structure your applications and transactions for maximum throughput.
+
+## How congestion control works
+
+Sui's congestion control algorithm runs every time a new batch of sequenced transactions is received from consensus.
+
+1. All transactions are **sorted in order of gas price**.
+
+1. Each transaction's **execution cost is estimated** using the [`TotalGasBudgetWithCap`](https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/authority/shared_object_congestion_tracker.rs#L143) heuristic.
+
+1. The [`SharedObjectCongestionTracker`](https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/authority/shared_object_congestion_tracker.rs) keeps a running tally of how much **per-object congestion budget** is used. If *all* shared objects used by the transaction have enough budget left, the transaction is scheduled. Otherwise it is **deferred** until the next commit.
+
+1. When a transaction is scheduled, it consumes budget for the *mutable* shared objects that it uses. (*Immutable* shared objects don't consume any budget, because multiple immutable uses of a shared object can be executed in parallel.)
```

Comment:
This feels like an add-on to the previous point rather than a standalone bullet (maybe right after "how much per-object congestion budget issued")